fix(tooltip): fix regression on touch devices with a mouse

crisbeto opened the pull request:
Reverts most of the changes from #8700 and switches to a simpler approach that uses a repeated `touchstart` to hide the tooltip, instead of `touchend`, which fires too early.
Also re-enables mouse events for all kinds of devices, in order to cover cases where the devices has both a touchscreen and a mouse.

Fixes #8710.

clshortfuse commented:
I believe you need to watch for ‘touchend‘ as well in LEAVE_EVENTS or the tooltip will get stuck. You can test this in Chrome with device emulation. Chrome will change your mouse events to touch events.

crisbeto commented:
I removed `touchend` on purpose, because it fires right after you lift your finger. This isn't particularly good UX, because it'll show the tooltip for a second and then remove it when you lift your finger. Additionally, this caused it to be a bit glitchy, because when the element isn't focused, the tooltip would get stuck, even though you lifted you finger. 

I switched it to open/close when you tap the tooltip target, in addition to the blur and on scroll from before.

clshortfuse commented:
Okay. Maybe there should be an autoclose timer for touch operations if it doesn't close automatically? It doesn't have to be part of this PR.

crisbeto commented:
I don't think that's an issue on a touch device, because you have 3 options:

1. You tapped the same target again, in which case it would close.
2. You tapped outside the target, in which case it fires the blur event and closes.
3. You scrolled away, in which case it closes.

clshortfuse commented:
Okay. Hopefully this doesn't break anything. I'll test it later. My tooltips don't have any delay set and I have plenty of md-icon-button buttons with tooltips. I don't want a tooltip to fire and not close every time a button a pressed.

crisbeto commented:
Well I retested on a proper touch device (I was mostly testing in the Chrome dev tools) and the blur event isn't quite as predictable. I'll figure something out for tapping out of it.
